aiohttp/client.py

```diff
@@ -267,12 +267,10 @@ def __init__(
         self._requote_redirect_url = requote_redirect_url
         self._read_bufsize = read_bufsize
 
-        # Convert to list of tuples
-        if headers:
-            real_headers: CIMultiDict[str] = CIMultiDict(headers)
-        else:
-            real_headers = CIMultiDict()
-        self._default_headers: CIMultiDict[str] = real_headers
+        # See `headers` property setter function. Mypy struggling with the property
+        # returning a stricter type than the setter function allows.
+        # https://github.com/python/mypy/issues/3004
+        self.headers = headers  # type: ignore
```

Comment:
Type ignores must never be wholesale. They should include the specific rule they are meant to suppress, and should not suppress all the rules existing now or in the future.